upload_vehicle_data.py

```diff
@@ -0,0 +1,322 @@
+        f"find {directory} -name '*.mcap' -type f -printf '%T+ %p\n' | sort | "
+        "head -n 1 | cut -d' ' -f2-"
+    )
+    try:
+        result = subprocess.run(
+            f'ssh {remote_user}@{remote_ip} "{list_cmd}"',
+            shell=True,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        oldest_file = result.stdout.strip()
+        if oldest_file:
+            delete_cmd = f'rm {oldest_file}'
+            run_ssh_command(remote_user, remote_ip, delete_cmd)
+            logging.info(f'Deleted oldest mcap file: {oldest_file}')
+    except subprocess.CalledProcessError as e:
+        logging.error(f'Failed to delete oldest mcap file: {e}')
+        raise
+
+
+def main(config):
+    """Automate the upload of rosbags."""
+    remote_user = config['remote_user']
+    remote_ip = config['remote_ip']
+    remote_directory = config['remote_directory']
+    remote_temp_directory = config['remote_temp_directory']
+    vdi_upload_directory = config['vdi_upload_directory']
+    # iperf_server_ip = config['iperf_server_ip']
+    clean_up = config['clean_up']
+    # Create the remote temporary directory
+    create_remote_directory(remote_user, remote_ip, remote_temp_directory)
+
+    try:
+        # Get the list of rosbags from the remote machine
+        rosbag_list = get_remote_rosbags_list(
+            remote_user, remote_ip, remote_directory
+        )
+
+        bandwidth_mbps = measure_bandwidth(remote_ip, remote_user)
+        if bandwidth_mbps is None:
+            print('Could not measure bandwidth. Exiting.')
+            return
+
+        total_size_gb = sum(
+            get_remote_file_size(remote_user, remote_ip, rosbag)
+            for rosbag in rosbag_list
+        )
+        estimated_time = get_estimated_upload_time(
+            total_size_gb, bandwidth_mbps
+        )
+
+        print(
+            f'Found {len(rosbag_list)} files to upload with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        print(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+        print(f'Estimated upload time: {estimated_time:.2f} hours.')
+        confirm = input('Do you want to proceed to upload? (yes/no): ')
+
+        if confirm.lower() != 'yes':
+            print('Upload aborted.')
+            return
+
+        logging.info(
+            f'Starting upload of {len(rosbag_list)} files with total size '
+            f'{total_size_gb:.2f} GB.'
+        )
+        logging.info(f'Measured bandwidth: {bandwidth_mbps:.2f} Mbps')
+
+        for rosbag in rosbag_list:
+            success = compress_and_transfer_rosbag(
+                remote_user,
+                remote_ip,
+                rosbag,
+                remote_temp_directory,
+                vdi_upload_directory,
+            )
+            if not success:
+                logging.error(
+                    f'Failed to process {rosbag}. Continuing with next file.'
+                )
+            else:

```

Comment:
Also, I think the current implementaiton will carry on with the next bag eventhoug the validation of the current rosbag uplaod (`rsync --checksum`) failed. We need to make sure all of the bags are uploaded so we need to retry uploading this file

Reply:
I changed the logic to make the function retry uploading the failed files. Because the `compress_and_transfer_rosbag` function will first check the available space on the vehicle, I think no space will not cause the success to be False.